src/xdsl/__init__.py

```diff
@@ -0,0 +1,38 @@
+_mlir_module = None
+_REQUIRED_MLIR_MODULES = ['ir']
+
+import types as _xdsl_init_types
+
+
+def load_mlir_module(mlir_module: _xdsl_init_types.ModuleType) -> None:
+    """
+    Configures xDSL to use the given MLIR module. This allows to use xDSL with
+    out-of-tree dialects, which ship with their own copy of the MLIR core
+    modules (see https://github.com/xdslproject/xdsl/issues/111). If so desired,
+    the user needs to call this function prior to importing `mlir_converter`.
+    Otherwise, `mlir` is used. Subsequent calls to this function are only
+    allowed if they are made with the same argument. Furthermore, the provided
+    module needs to be imported including all submodules specified in
+    _REQUIRED_MLIR_MODULES by the caller prior to calling this function.
+    """
+    global _mlir_module
+    if _mlir_module is not None and _mlir_module != mlir_module:
+        raise RuntimeError('Different modules already loaded previously.')
+    for submodule in _REQUIRED_MLIR_MODULES:
+        if not hasattr(mlir_module, submodule):
+            raise RuntimeError(
+                'Provided module "{}" does not have the submodule "{}" loaded.'
+                .format(mlir_module.__name__, submodule))
+    _mlir_module = mlir_module
+
+
+def ensure_loaded_mlir_module() -> None:
```

Comment:
I think this name should be `ensure_mlir_module_loaded`. `ensure_loaded_mlir_module` sounds like it ensures the loaded module, which I am not sure what it means but it sounds more like `ensure that this module is loaded` rather than `ensure that any module is loaded`. I think `ensure_mlir_module_loaded` makes this more explicit.

Reply:
OK, next commit renames the function.